--- boost_1_41_0_4.spec	2010-01-30 11:57:43.000000000 +0100
+++ boost_1_41_0_4.1.spec	2010-01-31 23:43:07.000000000 +0100
@@ -17,7 +17,7 @@
 Name: boost
 Summary: The free peer-reviewed portable C++ source libraries
 Version: 1.41.0
-Release: 4%{?dist}
+Release: 4.1%{?dist}
 License: Boost
 URL: http://sodium.resophonic.com/boost-cmake/%{version}.cmake0/
 Group: System Environment/Libraries
@@ -54,7 +54,6 @@
 # OpenMPI packages
 %if %{with openmpi}
 Requires: boost-openmpi = %{version}-%{release}
-Requires: boost-openmpi-devel = %{version}-%{release}
 Requires: boost-openmpi-python = %{version}-%{release}
 Requires: boost-graph-openmpi = %{version}-%{release}
 %endif
@@ -62,7 +61,6 @@
 # MPICH2 packages
 %if %{with mpich2}
 Requires: boost-mpich2 = %{version}-%{release}
-Requires: boost-mpich2-devel = %{version}-%{release}
 Requires: boost-mpich2-python = %{version}-%{release}
 Requires: boost-graph-mpich2 = %{version}-%{release}
 %endif
@@ -277,6 +275,8 @@
 %package openmpi-devel
 Summary: Shared library symlinks for Boost.MPI
 Group: System Environment/Libraries
+Requires: boost-devel = %{version}-%{release}
+Requires: boost-openmpi = %{version}-%{release}
 
 %description openmpi-devel
 
@@ -286,6 +286,7 @@
 %package openmpi-python
 Summary: Python runtime component of Boost.MPI library
 Group: System Environment/Libraries
+Requires: boost-openmpi = %{version}-%{release}
 
 %description openmpi-python
 
@@ -322,6 +323,8 @@
 %package mpich2-devel
 Summary: Shared library symlinks for Boost.MPI
 Group: System Environment/Libraries
+Requires: boost-devel = %{version}-%{release}
+Requires: boost-mpich2 = %{version}-%{release}
 
 %description mpich2-devel
 
@@ -331,6 +334,7 @@
 %package mpich2-python
 Summary: Python runtime component of Boost.MPI library
 Group: System Environment/Libraries
+Requires: boost-mpich2 = %{version}-%{release}
 
 %description mpich2-python
 
@@ -377,6 +381,9 @@
 # Build MPI parts of Boost with OpenMPI support
 %if %{with openmpi}
 %{_openmpi_load}
+# Work around the bug: https://bugzilla.redhat.com/show_bug.cgi?id=560224
+MPI_COMPILER=openmpi-%{_arch}
+export MPI_COMPILER
 ( echo ============================= build $MPI_COMPILER ==================
   mkdir $MPI_COMPILER
   cd $MPI_COMPILER
@@ -440,6 +447,9 @@
 
 %if %{with openmpi}
 %{_openmpi_load}
+# Work around the bug: https://bugzilla.redhat.com/show_bug.cgi?id=560224
+MPI_COMPILER=openmpi-%{_arch}
+export MPI_COMPILER
 echo ============================= install $MPI_COMPILER ==================
 DESTDIR=$RPM_BUILD_ROOT make -C $MPI_COMPILER VERBOSE=1 install
 # Remove parts of boost that we don't want installed in MPI directory.
@@ -507,6 +517,11 @@
 %clean
 %{__rm} -rf $RPM_BUILD_ROOT
 
+
+# MPI subpackages don't need the ldconfig magic.  They are hidden by
+# default, in MPI backend-specific directory, and only show to the
+# user after the relevant environment module has been loaded.
+
 %post date-time -p /sbin/ldconfig
 
 %postun date-time -p /sbin/ldconfig
@@ -559,9 +574,6 @@
 
 %postun wave -p /sbin/ldconfig
 
-# MPI subpackages don't need the ldconfig magic.  They are hidden by
-# default, in MPI backend-specific directory, and only show after the
-# relevant MPI rpm build macro is issued.
 
 
 %files
@@ -714,6 +726,9 @@
 %endif
 
 %changelog
+* Mon Jan 30 2010 Denis Arnaud <denis.arnaud_fedora@m4x.org> - 1.41.0-4.1
+- Various fixes on the specification
+
 * Mon Jan 25 2010 Petr Machata <pmachata@redhat.com> - 1.41.0-4
 - Add a patch to build mapnik
 - Resolves: #558383
