diff -up boost_1_53_0/boost/thread/future.hpp\~ boost_1_53_0/boost/thread/future.hpp
--- boost_1_53_0/boost/thread/future.hpp~	2012-12-16 20:01:45.000000000 +0100
+++ boost_1_53_0/boost/thread/future.hpp	2013-07-23 16:50:27.044995968 +0200
@@ -2911,10 +2913,6 @@ namespace boost
         {
           typedef typename boost::result_of<typename decay<F>::type()>::type R;
           typedef packaged_task<R()> packaged_task_type;
-
-          typedef detail::async_func<typename decay<F>::type> BF;
-          typedef typename BF::result_type Rp;
-
   #endif
 #else
         template <class F>
@@ -2925,9 +2924,6 @@ namespace boost
         {
           typedef typename boost::result_of<typename decay<F>::type()>::type R;
           typedef packaged_task<R> packaged_task_type;
-
-          typedef detail::async_func<typename decay<F>::type> BF;
-          typedef typename BF::result_type Rp;
 #endif
 
         if (int(policy) & int(launch::async))

Diff finished.  Tue Jul 23 16:50:30 2013
