diff -up boost_1_53_0/boost/lexical_cast.hpp\~ boost_1_53_0/boost/lexical_cast.hpp
--- boost_1_53_0/boost/lexical_cast.hpp~	2012-12-29 15:41:23.000000000 +0100
+++ boost_1_53_0/boost/lexical_cast.hpp	2013-07-22 23:27:48.841566845 +0200
@@ -865,7 +865,6 @@ namespace boost {
 #ifndef BOOST_NO_LIMITS_COMPILE_TIME_CONSTANTS
             BOOST_STATIC_ASSERT(!std::numeric_limits<T>::is_signed);
 #endif
-            typedef typename Traits::int_type int_type;
             CharT const czero = lcast_char_constants<CharT>::zero;
             --end;
             value = 0;

Diff finished.  Mon Jul 22 23:28:02 2013
