diff --git a/boost/property_tree/detail/json_parser_read.hpp b/boost/property_tree/detail/json_parser_read.hpp
index 3f06794..46acc5a 100644
--- a/boost/property_tree/detail/json_parser_read.hpp
+++ b/boost/property_tree/detail/json_parser_read.hpp
@@ -179,6 +179,9 @@ namespace boost { namespace property_tree { namespace json_parser
             {
 
                 using namespace boost::spirit::classic;
+                // There's a boost::assertion too, so another explicit using
+                // here:
+                using boost::spirit::classic::assertion;
 
                 // Assertions
                 assertion<std::string> expect_object("expected object");

