diff -urpN boost_1_44_0/libs/random/CMakeLists.txt boost_1_44_0-pm/libs/random/CMakeLists.txt
--- boost_1_44_0/libs/random/CMakeLists.txt	2011-01-03 16:52:25.000000000 +0100
+++ boost_1_44_0-pm/libs/random/CMakeLists.txt	2011-01-03 16:51:17.000000000 +0100
@@ -15,7 +15,7 @@ set (lib_headers
 # Add a library target to the build system
 boost_library_project(
   random
-  #  SRCDIRS 
+  SRCDIRS  src 
   TESTDIRS  test
   HEADERS ${lib_headers}
   #  DOCDIRS 
diff -urpN boost_1_44_0/libs/random/src/CMakeLists.txt boost_1_44_0-pm/libs/random/src/CMakeLists.txt
--- boost_1_44_0/libs/random/src/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ boost_1_44_0-pm/libs/random/src/CMakeLists.txt	2011-01-03 16:50:40.000000000 +0100
@@ -0,0 +1,7 @@
+add_definitions(-DBOOST_RANDOM_NO_LIB=1)
+boost_add_library(
+  random
+  random_device.cpp 
+  STATIC_COMPILE_FLAGS -DBOOST_RANDOM_STATIC_LINK
+  SHARED_COMPILE_FLAGS -DBOOST_ALL_DYN_LINK=1
+  )
